### PR TITLE
Pin gopkg versions, add grpc gateway and swagger support

### PR DESCRIPTION
**Pull request by generalhenry**
closes https://github.com/appcelerator/docker-protoc/issues

To test run with 
```
--go_out=Mgoogle/api/annotations.proto=github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis/google/api,plugins=grpc:/go/src/
--grpc-gateway_out=logtostderr=true:/go/src
--swagger_out=logtostderr=true:/go/src/
-I /go/src/
-I /go/src/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis
```

### CODE REVIEW

**Comment by subfuzion** [comment]
@generalhenry Please don't merge your own PRs.

File: Dockerfile [hunk 1]
```diff
@@ -1,26 +1,12 @@
...
-    apk del .builddeps && \
-    rm -rf /var/cache/apk/*
+RUN install-protoc
 
-# Get the source from GitHub
-RUN go get google.golang.org/grpc
-
-# Install protoc-gen-go
-RUN go get github.com/golang/protobuf/protoc-gen-go
+WORKDIR /go/src
```
Comment (subfuzion) [inline Dockerfile]: Duplicate of line 10.

File: Gofile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+google.golang.org/grpc 3235584c703acd8af8ab25132618c987c5606b66
```
Comment (subfuzion) [inline Gofile:1]: Do we need to pin on commits? I'd rather pin on the tag, eg, v1.0.1...

File: Dockerfile (hunk 1 above)
Comment (subfuzion) [inline Dockerfile:8]: Duplicate of line 10.

**Comment by subfuzion** [comment]
@ndegory You ever have a chance to review this? Be aware Henry's update pins down the grpc and related packages in `Gofile`.

**Comment by ndegory** [comment]
@subfusion, it has been tested while tagging as 0.3.0 (currently used by AMP), so far it works without issues, but I'm not a fan of having commit references instead of tags in Gofile. Which can't be fixed for protobuf, as they don't have tags on their repo...

**Pull request merged**